core/metadata.cpp

```diff
@@ -199,6 +222,187 @@ static bool parseMP4(QFile &f, metadata *metadata)
 	return found_ftyp;
 }
 
+static QStringList weekdays = { "mon", "tue", "wed", "thu", "fri", "sat", "sun" };
+static QStringList months = { "jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec" };
+
+static bool parseDate(const QString &s_in, timestamp_t &timestamp)
+{
+	// As a first attempt we're very crude: replace all '/' and '-' by ':'
+	// and try to see if this is of the form "yyyy:mm:dd hh:mm:ss".
+	// Since AVIs have no unified way of saving dates, we will have
+	// to find out empirically what different software produces.
+	// Note that we don't want to parse dates without time. That would
+	// be too imprecise and in such a case we'd rather go after the
+	// file modification date.
+	QString s = s_in;
+	s.replace('/', ':');
+	s.replace('-', ':');
+	QDateTime datetime = QDateTime::fromString(s, "yyyy:M:d h:m:s");
+	if (datetime.isValid()) {
+		// Not knowing any better, we suppose that time is give in UTC
+		datetime.setTimeSpec(Qt::UTC);
+		timestamp = datetime.toMSecsSinceEpoch() / 1000;
+		return true;
+	}
+
+	// I've also seen "Weekday Mon  Day hh:mm:ss yyyy"(!)
+	QStringList items = s.split(' ', QString::SkipEmptyParts);
+	if (items.size() < 4)
+		return false;
+
+	// Skip weekday if any is given
+	for (const QString &day: weekdays) {
+		if (items[0].startsWith(day, Qt::CaseInsensitive)) {
```

Comment:
so this is always encoded in English abbreviation format?
That's crazy

Reply:
I think there is no standardized way to record the creation date. As far as I understood, the creating software is basically free to write in any human-readable string it wants.

I implemented the two ways I found on the web. I'd really appreciate any real-world AVI examples. But I'm not sure if any modern video equipment produces such a thing.

Reply:
OK, makes sense. Let's go with what we have then.
I'd still appreciate the CHANGELOG entry :-)